### PR TITLE
fix: Consider using `super()` without arguments.

### LINKED ISSUE #26931: Consider using `super()` without arguments.
In many files I see that `super()` is being called builtin with the current class and instance. On Python 3 these arguments are the default and they can be omitted.
Examples:
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/numpy/device.py#L96
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/tensorflow/device.py#L108
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/paddle/device.py#L119
These arguments inside `super()` are by default from python 3, So they can be removed to make the code according to the latest standards.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description
In many files I see that `super()` is being called builtin with the current class and instance. On Python 3 these arguments are the default and they can be omitted.
Examples:
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/numpy/device.py#L96
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/tensorflow/device.py#L108
https://github.com/unifyai/ivy/blob/0d9e769776a48cbed15919fef7c123d824b547ff/ivy/functional/backends/paddle/device.py#L119
These arguments inside `super()` are by default from python 3, So they can be removed to make the code according to the latest standards.

## Related Issue
Closes #26931

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

**Comment by hello-fri-end** [comment]
Hey @Sai-Suraj-27, thanks for taking the time to look into this. I'll let @vedpatwardhan take the final call here with what fits best into our current design of these things.

**Comment by vedpatwardhan** [comment]
The changes look good to me, but given that the `README` and the `docs` were changed, let's wait to see what @guillesanbri and @KareemMAX think. Thanks @Sai-Suraj-27 😄

**Review by KareemMAX — APPROVED** [review]
LGTM! Waiting for @guillesanbri to approve

**Review by guillesanbri — APPROVED** [review]
lgtm!

**Pull request merged**
